src/exception/util.ts

```diff
@@ -17,11 +17,42 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { RepoError } from './exception-repo';
-import { ProgramException } from './types';
+import { RepoError } from './repo/types';
+import { EntityException, ProgramException, EntityValues, Entity } from './types';
+import _ from 'lodash';
 
-export function isProgramException(
-  result: ProgramException | RepoError,
-): result is ProgramException {
-  return (result as ProgramException).programId !== undefined;
+export function isProgramException(result: ProgramException | null): result is ProgramException {
```

Comment:
Question for my own TS understanding: why is `: result is ProgramException` necessary in this case?

Reply:
it's the predicate type guard syntax https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
🤔so it’s useful wherever the function returns, not within it.

got it! thanks 🙏